File: frontend/src/app/core/services/vacuna.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from './api.service';
import { ApiResponse, PagedResponse } from '../models/api-response.model';
import { Vacuna, VacunaRequest } from '../models/vacuna.model';

@Injectable({ providedIn: 'root' })
export class VacunaService {
  constructor(private api: ApiService) {}

  getAll(params?: any): Observable<ApiResponse<PagedResponse<Vacuna>>> {
    return this.api.getPaged('/vacunas', params?.page, params?.size, params);
  }

  getById(id: number): Observable<ApiResponse<Vacuna>> {
    return this.api.get(`/vacunas/${id}`);
  }

  create(data: VacunaRequest): Observable<ApiResponse<Vacuna>> {
    return this.api.post('/vacunas', data);
  }

  update(id: number, data: VacunaRequest): Observable<ApiResponse<Vacuna>> {
    return this.api.put(`/vacunas/${id}`, data);
  }

  delete(id: number): Observable<ApiResponse<void>> {
    return this.api.delete(`/vacunas/${id}`);
  }
}

File: backend/src/main/java/bo/capital/tec/pet/vacuna/service/VacunaService.java
package bo.capital.tec.pet.vacuna.service;

import bo.capital.tec.pet.common.response.PagedResponse;
import bo.capital.tec.pet.vacuna.dto.VacunaRequestDTO;
import bo.capital.tec.pet.vacuna.dto.VacunaResponseDTO;
import bo.capital.tec.pet.vacuna.dto.VacunaSummaryDTO;

public interface VacunaService {
    VacunaResponseDTO create(VacunaRequestDTO dto);
    VacunaResponseDTO getById(Long id);
    PagedResponse<VacunaSummaryDTO> getAll(String nombre, boolean activo, int page, int size);
    VacunaResponseDTO update(Long id, VacunaRequestDTO dto);
    void delete(Long id);
}
